spec.md

```diff
@@ -451,8 +462,15 @@ encapsulated within the `data` attribute.
 
 - Type: `Any`
 - Description: The event payload. The payload depends on the `type` and the
-  `schemaurl`. It is encoded into a media format which is specified by the
-  `datacontenttype` attribute (e.g. application/json).
+  `schemaurl`. If the type is `Binary` or `String`, it is encoded using a 
+  media format which is specified by the `datacontenttype` attribute 
+  (e.g. `application/json`).
+
+  The abstract type (ie, `Binary`, `Text`, `Integer` or `Map`) MAY change when
```

Comment:
We agree on the effect; I've tried to explain this differently in the proposal above ("The attribute MAY be set for all other types..."). What happens is that the in-memory structure at the publisher might contain a map, but `datacontenttype` is set to "application/json" while the event is encoded as Avro. In that case, JSON rules apply to `data` and the attribute ends up being a String holding JSON inside the Avro-encoded event and therefore pops out like that on the other end, but I don't think that ought to be a real surprise. We might write a primer paragraph that clarifies that behavior.    

When you use the normative "MAY", you're permitting something, but here you're actually just describing a potential effect, which is a "might".